src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.cpp

```diff
@@ -0,0 +1,236 @@
+                                           true);
+
+    auto activeSequenceContext = annDnaView->getActiveSequenceContext();
+    SAFE_POINT(activeSequenceContext != nullptr, L10N::nullPointerError("ADVSequenceObjectContext"), );
+
+    auto seqLength = activeSequenceContext->getSequenceLength();
+
+    sbLeftAreaStart->setValue(1);
+    sbLeftAreaEnd->setValue(seqLength / 10);
+    sbRightAreaStart->setValue(seqLength / 10 * 5);
+    sbRightAreaEnd->setValue(seqLength / 10 * 6);
+
+    connect(pbStart, &QAbstractButton::clicked, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_start);
+    connect(tbLeftAreaSelectManually, &QAbstractButton::clicked, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_selectManually);
+    connect(tbRightAreaSelectManually, &QAbstractButton::clicked, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_selectManually);
+
+    U2WidgetStateStorage::restoreWidgetState(savableWidget);
+
+    sl_updateMaxValues();
+    for (auto minSb : minMaxValuesSb.keys()) {
+        connect(minSb, &QSpinBox::editingFinished, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_minValueChanged);
+        connect(minMaxValuesSb[minSb], &QSpinBox::editingFinished, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_maxValueChanged);
+    }
+    connect(tbLoadBackbone, &QAbstractButton::clicked, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_loadBackbone);
+    connect(tbsaveRandomSequences, &QAbstractButton::clicked, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_saveRandomSequences);
+    connect(tbLoadOtherSequencesInPcr, &QAbstractButton::clicked, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_loadOtherSequenceInPcr);
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_start() {
+    //TODO
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_selectManually() {
+    auto currentSMButton = qobject_cast<QToolButton*>(sender());
+    SAFE_POINT(currentSMButton != nullptr, L10N::nullPointerError("QAbstractButton"), );
+
+    if (smButton != nullptr) {
+        bool disableState = true;
+        if (smButton->isChecked()) {
+            smButton->setChecked(false);
+            disableState = false;
+        }
+        smButton = nullptr;
+        start = nullptr;
+        end = nullptr;
+        disconnect(updateRegionConnection);
+        updateRegionConnection = QMetaObject::Connection();
+
+        if (disableState) {
+            return;
+        }
+    }
+
+    smButton = currentSMButton;
+    if (smButton == tbLeftAreaSelectManually) {
+        start = sbLeftAreaStart;
+        end = sbLeftAreaEnd;
+    } else {
+        start = sbRightAreaStart;
+        end = sbRightAreaEnd;
+    }
+    auto activeSequenceContext = annDnaView->getActiveSequenceContext();
+    SAFE_POINT(activeSequenceContext != nullptr, L10N::nullPointerError("ADVSequenceObjectContext"), );
+
+    auto sequenceSelection = activeSequenceContext->getSequenceSelection();
+    SAFE_POINT(sequenceSelection != nullptr, L10N::nullPointerError("DNASequenceSelection"), );
```

Comment:
same here: it should not be possible for selection to be a nullptr

Reply:
OK